Lib/distutils/util.py

```diff
@@ -92,6 +94,23 @@ def get_platform ():
 
 # get_platform ()
 
+def get_target_platform ():
```

Comment:
If this needs to replace most uses of `get_platform` except when detecting whether we are cross-compiling, can we just update `get_platform` (and avoid all the renames above) and provide a new API for generating the tag?

On second reading, we probably aren't properly handling the `--plat-name` option in `build_ext` properly already, and that would be the better way to specify that "arm" is the target. It can select a default based on the environment variable, but it should also work when passed `-p win-arm` and going via the `vcvarsall.bat` path.

Reply:
get_target_platform returns the target platform when cross-compiling.
get_platform returns the host platform.
ARM32 binaries can only be cross-compiled using the MSVC toolset.
Also when a module like numpy overrides the build files --target-plat doesn't get always passed through in the attributes like it should

Reply:
According to the [docs](https://docs.python.org/3/distutils/apiref.html?highlight=get_platform#distutils.util.get_platform), it already returns the target platform on macOS, as well as already being wrong for Windows ("non-POSIX platforms … return `sys.platform`"), so I think changing it on Windows to also return the distutils target is fine.

Reply:
I'll start a post on python-dev to clarify and see what people think.

Reply:
I updated the PR to use get_platform and get_host_platform.  Testing is still needed